### PR TITLE
Update to support building native linux binaries

### PR DESCRIPTION
**Pull request by finomen**

### CODE REVIEW

File: RULES/cc/toolchain.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ type ToolchainFlavor int
 const (
 	Linux ToolchainFlavor = iota
 	Windows
+	LinuxCxx
```
[the remark this answers is not in the document]
Comment (finomen) [reply RULES/cc/toolchain.go:16]: This flag is used only for link rules

File: RULES/cc/cc.go [hunk 2]
```diff
@@ -481,18 +481,20 @@ func (inputLibrary Library) CcLibrary(toolchain Toolchain) Library {
...
+	AsFlags         []string
+	LinkerFlags     []string
+	LinkerFlagsPost []string
+	Deps            []Dep
+	DepsPre         []Dep
+	DepsPost        []Dep
+	Script          core.Path
+	Toolchain       Toolchain
+	Includes        []core.Path
+	Objects         []core.Path
```
Comment (Javier-varez) [inline RULES/cc/cc.go:497]: nit: Objs to keep the same name as in libs

File: RULES/cc/toolchain.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Javier-varez) [reply RULES/cc/toolchain.go:16]: Then it is probably worth renaming ;)

File: RULES/cc/toolchain.go [hunk 3]
```diff
@@ -8,11 +8,14 @@ import (
...
 
-type ToolchainFlavor int
+type LinkerFlavor int
 
 const (
-	Linux ToolchainFlavor = iota
-	Windows
+	Ld LinkerFlavor = iota
+	LdLld
+	LdLink
```
Comment (Javier-varez) [inline RULES/cc/toolchain.go:16]: Shouldn't this be `LldLink`?

**Review by Javier-varez — APPROVED** [review]

**Pull request merged**